### PR TITLE
Document All Capability Atoms and Profiles

### PR DESCRIPTION
**Pull request by ArielG-NV**
Fixes: #4125

Unimplemented Considerations:
1. This PR does not add support to query all capability-atom's from a command-line option. It is understood that this might be desired, due to this, the logic to generate `docs\user-guide\a3-02-reference-capability-atoms.md` was made to be "command-line friendly" so minimal changes are needed to pipe our documentation into a command-line option if this change is to be added.

Changes:
1. Added a way to document atoms inside `.capdef`. Method to document is described under `source\slang\slang-capabilities.capdef`. The goal is to error if a public atom does not have any form of documentation to ensure we always have up-to-date documentation to guide user on what an atom is/does. 
    * The following `.capdef` file syntax was added 
        * /// [HEADER_GROUP] 
        * /// regular comment
3. When capability generator runs it auto-generates `docs\user-guide\a3-02-reference-capability-atoms.md`
4. Added to the user-guide 3 sections: `Reference`, `Reference -> Capability Profiles`, `Reference -> Capability atoms` section

### CODE REVIEW

File: docs/user-guide/a3-02-reference-capability-atoms.md [hunk 1]
```diff
@@ -0,0 +1,2329 @@
...
+3. [Versions](#Versions)
+4. [Extensions](#Extensions)
+5. [Compound Capabilities](#Compound-Capabilities)
+6. [Other](#Other)
+
+<br>
+
+Targets
+----------------------
+*Capabilities to specify code generation targets (`glsl`, `spirv`...)*<br><br>
```
Comment (csyonghe) [inline docs/user-guide/a3-02-reference-capability-atoms.md:21]: don't use `<br>`, just two empty lines.

And it seems that there are too many extraneous empty lines here.

**Review by csyonghe — APPROVED** [review]

**Comment by csyonghe** [comment]
Looks like there are still `<br>` in the generated md.

**Comment by ArielG-NV** [comment]
> Looks like there are still <br> in the generated md.

I did not realize we wanted to remove all `<br>`s, I will make a new commit without any `<br>` generation.

**Review by csyonghe — APPROVED** [review]

**Comment by jkwak-work** [comment]
It is strange to see that the PR description describes what it doesn't do and it is missing what it does.

**Comment by csyonghe** [comment]
The `Changes:` section is about what it is doing?

**Comment by jkwak-work** [comment]
I think it would be nicer if each item in the generated document was sorted in an alphabet order because there are so many of them and it is difficult to navigate or find one.

**Comment by ArielG-NV** [comment]
> I think it would be nicer if each item in the generated document was sorted in an alphabet order because there are so many of them and it is difficult to navigate or find one.

Issue has been made to track this (https://github.com/shader-slang/slang/issues/5030)

**Pull request merged**